plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/lookup/ShellLookup.java

```diff
@@ -142,50 +146,31 @@ public Shell run() {
 	/**
 	 * Waits for specified time period for a shell matching specified matcher.
 	 * 
-	 * @param matcher matcher to match shell
+	 * @param matchers matchers to match shell
 	 * @param timePeriod time period to wait for
 	 * @return shell matching specified matcher
 	 */
-	public Shell getShell(final Matcher<String> matcher , TimePeriod timePeriod) {
-		if (!timePeriod.equals(TimePeriod.NONE)){
-			new WaitUntil(new ShellMatchingMatcherIsAvailable(matcher), timePeriod, false);
+	public Shell getShell(TimePeriod timePeriod, Matcher<?>... matchers) {
+		ShellMatchingMatcherIsAvailable shellCondition = new ShellMatchingMatcherIsAvailable(matchers);
+		try{
+			new WaitUntil(shellCondition, timePeriod);
+		}catch (WaitTimeoutExpiredException e) {
+			AndMatcher am  = new AndMatcher(matchers);
```

Comment:
just a though. Almost every wait condition has some matcher, maybe we could create a layer in hierarchy of wait conditions with method on a wait condition getMatcher() which would return matcher(s). Then we could call in such places just shellCondition.getMatcher() instead of instantiating it this way - because now you had to know that the ShellMatchinMatcherIsAvailable internally uses AndMatcher. We would have it more transparent.